includes/api/class-wc-rest-product-variations-controller.php

```diff
@@ -248,11 +249,17 @@ protected function prepare_object_for_database( $request, $creating = false ) {
 
 		$variation->set_parent_id( absint( $request['product_id'] ) );
 
-		// Status.
+		// TODO Fix/Remove in future API version bump. This controls the wrong thing.
+		// https://github.com/woocommerce/woocommerce/issues/15215
 		if ( isset( $request['visible'] ) ) {
 			$variation->set_status( false === $request['visible'] ? 'private' : 'publish' );
 		}
 
+		// Status.
```

Comment:
I'm not sure I agree.. we never did that on WordPress.com or anything else I have worked on. Adding a field doesn't break backwards compatibility. Changing or removing a field does.